### PR TITLE
Add Go AES-GCM tests.

### PR DESCRIPTION
**Pull request by jhoyla**
This is a preliminary piece of work, beginning to apply Wycheproof to Golang's crypto library.

Some tests fail for minor issues, but I thought it more important to mirror the Java tests accurately than to achieve no failures. 

Tests that fail:

1. AllVectors - The Go crypto library allows you to force a non-standard sized nonce. If forced, it will allow a zero size nonce. 
2. IVReuse - The Go crypto library does not track IVs, and thus allows IV reuse. 
3. ByteArrayTooShort - Go does not throw an error on writing to an array that is too short, the write fails silently. This behaviour differs from the Java behaviour. 
4. EncryptWithPlaintextWithEmptyIv (forced) - See 1.
5. DecryptWithEmptyIv (forced) - See 1.

### CODE REVIEW

**Comment by jhoyla** [comment]
I signed it.

File: BUILD.bazel [hunk 1]
```diff
@@ -190,3 +190,14 @@ closure_js_deps(
...
     deps = ["@google_e2e//:e2e"],
 )
+
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library", "go_test")
+#load("@bazel_gazelle//:def.bzl", "gazelle")
+
+# Go Cipher tests
+go_test(
+    name = "go_default_test",
+    srcs = ["go/src/github.com/jhoyla/wycheproof/aes_gcm_test.go"],
```
Comment (cemeyer) [inline BUILD.bazel]: Should this point at google/wycheproof for the PR?

**Comment by thaidn** [comment]
Wow, I didn't notice this thread. Will take a look closely and merge.

**Comment by henrydcase** [comment]
It looks interesting. Any hope to get it merged?

**Comment by thaidn** [comment]
We'll take a look. In order to merge something, our internal tooling requires us to sync internal changes to here first, but we're blocked by a logistic issue that prevents us from updating this repo.

Stay tuned!

**Comment by thaidn** [comment]
The logistic issue was resolved. I'm trying to merge this by the end of this week.

**Comment by thaidn** [comment]
This PR is on hold because it's not clear atm that we'd have resources to maintain new test drivers.

**Comment by Jojojeto** [comment]
G

**Comment by Jojojeto** [comment]
G

**Comment by chuckx** [comment]
Following up after a long while... I'm going to close this PR as maintaining additional drivers is out of scope, as alluded to previously.

If you haven't already seen it, you may be interested in https://cs.opensource.google/go/x/crypto/+/master:internal/wycheproof/.

Those tests include coverage for a large chunk of the Go crypto libraries. For reference., the AES-GCM tests can be found [here](https://cs.opensource.google/go/x/crypto/+/master:internal/wycheproof/aead_test.go;l=125-135;drc=7067223927c4e3f3bb91a5c6e0d2aae83df74e7a) alongside other AEADs.

**Pull request closed without merging**